src/data/zmod/basic.lean

```diff
@@ -443,6 +443,20 @@ begin
   rw [←zmod.int_coe_eq_int_coe_iff', int.cast_coe_nat, zmod.nat_cast_val, zmod.cast_id],
 end
 
+lemma zmod.coe_neg_one {R : Type*} [ring R] (n : ℕ) : ↑(-1 : zmod n) = (n - 1 : R) :=
```

Comment:
Note we already have `cast_neg`, a weaker statement. It might be a good idea to spell out the difference in a docstring.

Reply:
Can you prove this for `(-1).val` first using `fin.coe_neg`?

Reply:
That is, something like:
```lean
lemma zmod.val_neg_one {R : Type*} [ring R] (n : ℕ) : (-1 : zmod n.succ).val = n :=
begin
  rw [val, fin.coe_neg],
  cases n,
  { rw [nat.mod_one] },
  { rw [fin.coe_one, nat.succ_add_sub_one, nat.mod_eq_of_lt (nat.lt.base _)], },
end
```

Reply:
Thanks!